addons/project_purchase/models/project.py

```diff
@@ -135,7 +135,7 @@ def _get_profitability_items(self, with_action=True):
                 **pol,
                 'invoice_lines': self.env['purchase.order.line'].browse(pol['id']).sudo().invoice_lines,  # One2Many cannot be queried, they are not columns
             } for pol in self._cr.dictfetchall()]
-            purchase_order_line_invoice_line_ids = self._get_already_included_profitability_invoice_line_ids()
+            purchase_order_line_invoice_line_ids = self.sudo()._get_already_included_profitability_invoice_line_ids()
```

Comment:
Actually, I would add the sudo only on the problematic search, that is the search made in project_hr_expense module.

Here is the diff. 🙂 

```diff
diff --git a/addons/project_hr_expense/models/project.py b/addons/project_hr_expense/models/project.py
index 2956fa7fd1fb..0e491d966391 100644
--- a/addons/project_hr_expense/models/project.py
+++ b/addons/project_hr_expense/models/project.py
@@ -72,7 +72,7 @@ class Project(models.Model):
         # As both purchase orders and expenses (paid by employee) create vendor bills,
         # we need to make sure they are exclusive in the profitability report.
         move_line_ids = super()._get_already_included_profitability_invoice_line_ids()
-        query = self.env['account.move.line']._search([
+        query = self.env['account.move.line'].sudo()._search([
             ('move_id.expense_sheet_id', '!=', False),
             ('id', 'not in', move_line_ids),
         ])
diff --git a/addons/project_purchase/models/project.py b/addons/project_purchase/models/project.py
index 5b4f5f7db308..930d408cefa2 100644
--- a/addons/project_purchase/models/project.py
+++ b/addons/project_purchase/models/project.py
@@ -135,7 +135,7 @@ class Project(models.Model):
                 **pol,
                 'invoice_lines': self.env['purchase.order.line'].browse(pol['id']).sudo().invoice_lines,  # One2Many cannot be queried, they are not columns
             } for pol in self._cr.dictfetchall()]
-            purchase_order_line_invoice_line_ids = self.sudo()._get_already_included_profitability_invoice_line_ids()
+            purchase_order_line_invoice_line_ids = self._get_already_included_profitability_invoice_line_ids()
             if purchase_order_line_read:

                 # Get conversion rate from currencies to currency of the project
```

Reply:
@xavierbol Thank you, i have updated the PR.